packages/php/email-editor/src/Integrations/Core/Renderer/Blocks/class-table.php

```diff
@@ -0,0 +1,840 @@
+		$allowed = array( 'solid', 'dashed', 'dotted' ); // Email-safe subset.
+		return in_array( $style, $allowed, true ) ? $style : 'solid';
+	}
+
+	/**
+	 * Add thicker borders for table headers and footers when no custom border is set.
+	 *
+	 * @param \WP_HTML_Tag_Processor $html HTML tag processor.
+	 * @param string                 $base_styles Base cell styles.
+	 * @param string                 $border_color Border color.
+	 * @param string                 $current_section Current table section (thead, tbody, tfoot).
+	 * @param string|null            $custom_border_width Custom border width if set.
+	 * @return string Updated cell styles.
+	 */
+	private function add_header_footer_borders( \WP_HTML_Tag_Processor $html, string $base_styles, string $border_color, string $current_section = '', ?string $custom_border_width = null ): string {
+		$tag_name = $html->get_tag();
+
+		// Only add thicker borders if no custom border width is set.
+		if ( $custom_border_width ) {
+			return $base_styles;
+		}
+
+		// Add thicker bottom border to all TH elements (headers).
+		if ( 'TH' === $tag_name ) {
+			$base_styles .= "; border-bottom: 3px solid {$border_color};";
+		}
+
+		// Add thicker top border to footer cells (TD elements in tfoot).
+		if ( 'TD' === $tag_name && 'tfoot' === $current_section ) {
+			$base_styles .= "; border-top: 3px solid {$border_color};";
+		}
+
+		return $base_styles;
+	}
+
+	/**
+	 * Get text alignment for a table cell.
+	 *
+	 * @param \WP_HTML_Tag_Processor $html HTML tag processor.
+	 * @return string Text alignment value (left, center, right).
+	 */
+	private function get_cell_text_alignment( \WP_HTML_Tag_Processor $html ): string {
+		// Check for data-align attribute first.
+		$data_align = $html->get_attribute( 'data-align' );
+		if ( $data_align && in_array( $data_align, self::VALID_TEXT_ALIGNMENTS, true ) ) {
+			return $data_align;
+		}
+
+		// Check for has-text-align-* classes.
+		$class_attr = (string) ( $html->get_attribute( 'class' ) ?? '' );
+		if ( false !== strpos( $class_attr, 'has-text-align-center' ) ) {
+			return 'center';
+		}
+		if ( false !== strpos( $class_attr, 'has-text-align-right' ) ) {
+			return 'right';
+		}
+		if ( false !== strpos( $class_attr, 'has-text-align-left' ) ) {
+			return 'left';
+		}
+
+		// Default to left alignment.
+		return 'left';
+	}
+
+	/**
+	 * Check if table has fixed layout class.
+	 *
+	 * @param string $class_attr Class attribute string.
+	 * @return bool True if has-fixed-layout class is present.
+	 */
+	private function has_fixed_layout( string $class_attr ): bool {
+		return false !== strpos( $class_attr, 'has-fixed-layout' );
+	}
+
+	/**
+	 * Extract table content and caption from figure wrapper if present.
+	 *
+	 * @param string $block_content Block content.
+	 * @return array Array with 'table' and 'caption' keys.
+	 */
+	private function extract_table_and_caption_from_figure( string $block_content ): array {
```

Comment:
I haven't looked into it too closely, but I wonder if the figcaption for a table works any differently from the one used in an [image block](https://github.com/woocommerce/woocommerce/blob/4635dc972b08e1b23db87de8d73eb8014f9877c6/packages/php/email-editor/src/Integrations/Core/Renderer/Blocks/class-image.php#L37). If they they overlap, could we share the the code for handling the figcaption?